### PR TITLE
Text Filtering additions

### PR DESCRIPTION
**Pull request by liach**
Gonna get a mod that allows me to turn off filters. Just overwrite method_33883

Fyi I didn't want to investigate this (and am not a psycho); I only saw there is an unmapped inner class for TextStream

### CODE REVIEW

**Comment by liach** [comment]
The most shocking aspect of 21w07a is how they added all these censorship stuff without zero mentions in change log or slicelime video. I feel gloomy!

File: mappings/net/minecraft/block/entity/SignBlockEntity.mapping [hunk 1]
```diff
@@ -24,3 +27,14 @@ CLASS net/minecraft/class_2625 net/minecraft/block/entity/SignBlockEntity
...
+		ARG 2 filtered
+	METHOD method_33384 unparsedTextFromJson (Ljava/lang/String;)Lnet/minecraft/class_2561;
+		ARG 1 json
+	METHOD method_33827 setTextOnRow (ILnet/minecraft/class_2561;Lnet/minecraft/class_2561;)V
+		ARG 1 row
+		ARG 2 text
+		ARG 3 filteredText
+	METHOD method_33828 parseTextFromJson (Ljava/lang/String;)Lnet/minecraft/class_2561;
+		ARG 1 json
+	METHOD method_33830 getTexts (Z)[Lnet/minecraft/class_2561;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/entity/SignBlockEntity.mapping:39]:
```suggestion
	METHOD method_33830 getText (Z)[Lnet/minecraft/class_2561;
```

File: mappings/net/minecraft/block/entity/SignBlockEntity.mapping [hunk 2]
```diff
@@ -24,3 +27,14 @@ CLASS net/minecraft/class_2625 net/minecraft/block/entity/SignBlockEntity
 		ARG 1 value
 	METHOD method_30843 getTextOnRow (IZ)Lnet/minecraft/class_2561;
 		ARG 1 row
+		ARG 2 filtered
+	METHOD method_33384 unparsedTextFromJson (Ljava/lang/String;)Lnet/minecraft/class_2561;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/entity/SignBlockEntity.mapping:31]:
```suggestion
	METHOD method_33384 unparseTextFromJson (Ljava/lang/String;)Lnet/minecraft/class_2561;
```

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/block/entity/SignBlockEntity.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/block/entity/SignBlockEntity.mapping:39]: Hmm, this returns an array of text, either 4 lines of text1..4 or filteredtext1..4 dependent on the parameter

File: mappings/net/minecraft/block/entity/SignBlockEntity.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/block/entity/SignBlockEntity.mapping:31]: This text returned is just like text from json, but the resulting score/nbt text etc aren't immediately resolved. Imo unparsed is better than unparse, and feel free to post more suggstions

Comment (YanisBft) [reply mappings/net/minecraft/block/entity/SignBlockEntity.mapping:31]: Make sense then 👍

File: mappings/net/minecraft/block/entity/SignBlockEntity.mapping (hunk 1 above)
Comment (YanisBft) [reply mappings/net/minecraft/block/entity/SignBlockEntity.mapping:39]: The `text` field needs to be renamed to follow this method's name

File: mappings/net/minecraft/block/entity/SignBlockEntity.mapping [hunk 3]
```diff
@@ -4,6 +4,9 @@ CLASS net/minecraft/class_2625 net/minecraft/block/entity/SignBlockEntity
 	FIELD field_12049 textBeingEdited [Lnet/minecraft/class_5481;
 	FIELD field_12050 text [Lnet/minecraft/class_2561;
```
Comment (liach) [inline mappings/net/minecraft/block/entity/SignBlockEntity.mapping]:
```suggestion
	FIELD field_12049 textsBeingEdited [Lnet/minecraft/class_5481;
	FIELD field_12050 texts [Lnet/minecraft/class_2561;
```

File: mappings/net/minecraft/block/entity/SignBlockEntity.mapping [hunk 4]
```diff
@@ -4,6 +4,9 @@ CLASS net/minecraft/class_2625 net/minecraft/block/entity/SignBlockEntity
...
 	FIELD field_16419 textColor Lnet/minecraft/class_1767;
+	FIELD field_28905 TEXT_KEYS [Ljava/lang/String;
+	FIELD field_28906 FILTERED_TEXT_KEYS [Ljava/lang/String;
+	FIELD field_28907 filteredText [Lnet/minecraft/class_2561;
```
Comment (liach) [inline mappings/net/minecraft/block/entity/SignBlockEntity.mapping]:
```suggestion
	FIELD field_28907 filteredTexts [Lnet/minecraft/class_2561;
```

**Review by YanisBft — APPROVED** [review]

**Comment by liach** [comment]
Given that other people start to map these stuff and cause conflicts, I request reviews on these.

**Review by enbrain — APPROVED** [review]

**Comment by modmuss50** [comment]
This failed as the first time I ran it, there were merge conflicts. I will look into hanlding this error and possibly adding the outdated label.

Either way there are merge conflicts on this PR.

**Review by enbrain — APPROVED** [review]

**Pull request merged**